src/Functions/FunctionsConversion.h

```diff
@@ -147,22 +177,53 @@ struct ConvertImpl
                     else if constexpr (IsDataTypeNumber<FromDataType> && IsDataTypeDecimal<ToDataType>)
                         vec_to[i] = convertToDecimal<FromDataType, ToDataType>(vec_from[i], vec_to.getScale());
                     else
-                        throw Exception("Unsupported data type in conversion function", ErrorCodes::CANNOT_CONVERT_TYPE);
+                    {
+                        if constexpr (convert_strategy == ConvertStrategy::Accurate)
+                            vec_null_map_to[i] = true;
+                        else
+                            throw Exception("Unsupported data type in conversion function", ErrorCodes::CANNOT_CONVERT_TYPE);
+                    }
                 }
                 else if constexpr (is_big_int_v<FromFieldType> || is_big_int_v<ToFieldType>)
                 {
                     if constexpr (std::is_same_v<FromFieldType, UInt128> || std::is_same_v<ToFieldType, UInt128>)
-                        throw Exception("Unexpected UInt128 to big int conversion", ErrorCodes::NOT_IMPLEMENTED);
+                    {
+                        if constexpr (convert_strategy == ConvertStrategy::Accurate)
+                            vec_null_map_to[i] = true;
+                        else                 
+                            throw Exception("Unexpected UInt128 to big int conversion", ErrorCodes::NOT_IMPLEMENTED);
+                    }
                     else
                         vec_to[i] = bigint_cast<ToFieldType>(vec_from[i]);
                 }
                 else if constexpr (std::is_same_v<ToFieldType, UInt128> && sizeof(FromFieldType) <= sizeof(UInt64))
                     vec_to[i] = static_cast<ToFieldType>(static_cast<UInt64>(vec_from[i]));
                 else
-                    vec_to[i] = static_cast<ToFieldType>(vec_from[i]);
+                {
+                    if constexpr (convert_strategy == ConvertStrategy::Accurate)
+                    {
+                        /// TODO: Add support for big integers
+                        if (accurate::greaterOp(std::numeric_limits<ToFieldType>::max(), vec_from[i]) ||
```

Comment:
@alexey-milovidov not sure how to properly implement it for types without numeric limits, do we have some places where similar functionality implemented ? Thanks.

Reply:
The most simple is to implement numeric limits for bigints.